packages/keystone/healthCheck.js

```diff
@@ -111,6 +115,70 @@ const HEALTHCHECK_OK = 200
 const HEALTHCHECK_ERROR = 500
 const HEALTHCHECK_WARNING = 417
 
+/**
+ * Creates a health check function that verifies the remaining GraphQL rate limit complexity.
+ * Intended to suppress duplicate alerts across pods by using a shared Redis key.
+ *
+ * @param {Object} options - Configuration options.
+ * @param {string} options.serviceName - Name to appear in health check results.
+ * @param {string} options.endpoint - GraphQL endpoint to be queried.
+ * @param {Object} options.authRequisites - Credentials or data used for authorization.
+ * @param {number} [options.suppressDurationInSeconds=3600] - Duration (in seconds) to suppress repeated failures after the first one is detected.
+ */
+const getRateLimitHealthCheck = ({
+    serviceName,
+    endpoint,
+    authRequisites,
+    suppressDurationInSeconds = 3600,
+}) => {
+    if (!serviceName) throw new Error('Service name must be provided')
+    if (!endpoint) throw new Error('GraphQL endpoint must be provided')
+    if (!authRequisites) throw new Error('Authentication requisites must be provided')
+
+    const redisKey = serviceName + '-rate-limit-healthcheck'
+
+    return {
+        name: `${serviceName}_rate_limit`,
+
+        prepare: async () => {
+            this.redisClient = getKVClient()
+            this.client = new ApolloServerClient(endpoint, authRequisites, {
+                clientName: 'healthcheck-client',
+            })
+        },
+
+        run: async () => {
+            try {
+                const isSuppressed = await this.redisClient.get(redisKey)
+                if (isSuppressed) {
+                    return PASS
+                }
+
+                await this.client.executeAuthorizedQuery({
+                    query: gql`{ __typename }`,
+                })
+
+                return PASS
+
+            } catch (error) {
```

Comment:
What about returning `WARN` when the used quota is close to the limit?

Reply:
It's ok. We need to see error and to write code to avoid this error
There is no need to check headers of the request

Reply:
A health check aims to control some limits and create alerts if there is too close to the limit or the limit is reached.
It's better to use logs for searching places of code for refactoring.